atoma-bin/atoma_node.rs

```diff
@@ -153,6 +153,24 @@ async fn initialize_tokenizers(
     try_join_all(fetch_futures).await
 }
 
+fn spawn_with_shutdown<F, E>(
+    f: F,
+    shutdown_sender: watch::Sender<bool>,
+) -> tokio::task::JoinHandle<Result<()>>
+where
+    E: Into<anyhow::Error>,
+    F: std::future::Future<Output = Result<(), E>> + Send + 'static,
+{
+    tokio::task::spawn(async move {
+        let res = f.await;
+        if res.is_err() {
+            // Only send shutdown signal if the task failed
+            shutdown_sender.send(true).unwrap();
```

Comment:
Can we please handle error here ?

Reply:
why should we handle it here, it's handled elsewhere, this just propagates the error